addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1073,6 +1116,43 @@ def create_group(self, partners_to, default_display_mode=False, name=''):
         channel._broadcast(channel.channel_member_ids.partner_id.ids)
         return channel
 
+    def _create_sub_channel(self, from_message_id=None, name=None):
+        self.ensure_one()
+        message = self.env["mail.message"]
+        if from_message_id:
+            message = self.env["mail.message"].search([("id", "=", from_message_id)])
```

Comment:
And/or maybe also with a constraint, as it could be used to leak data otherwise if we fetch stuff as sudo sometimes and it includes the from_message_id field.

Reply:
Constraint+test added.